include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -115,38 +115,37 @@ namespace VectorTools
     // vectorfunction multiple times at the
     // same point on a cell.
     //
-    // note that we have to set up all of the
-    // following arrays for each of the
-    // elements in the FECollection (which
-    // means only once if this is for a regular
-    // DoFHandler)
-    std::vector<std::vector<Point<dim> > > unit_support_points (fe.size());
+    // First check that the desired components are interpolating.
     for (unsigned int fe_index=0; fe_index<fe.size(); ++fe_index)
       {
-        unit_support_points[fe_index] = fe[fe_index].get_unit_support_points();
-        Assert ((unit_support_points[fe_index].size() != 0)||(fe[fe_index].dofs_per_cell == 0),
-                ExcNonInterpolatingFE());
-      }
+        for (unsigned int component_index = 0; component_index < n_components; ++component_index)
+          {
+            if (component_mask[component_index] == true)
+              {
+                std::pair<unsigned int, unsigned int> base_index = fe[fe_index].component_to_base_index(component_index);
 
+                AssertThrow ((fe[fe_index].base_element(base_index.first).has_support_points()) ||
```

Comment:
Fixed base_index by inlining.

For the Assert, I think that it should be AssertThrow since while not dependent on runtime state, it is a criterion that will result in a specific failure mode that is most easily tested at this point, and this is the correct failure message. AssertThrow also appears to be required for my failure test logic to work.